### PR TITLE
Add the ability to give a computer keyboard focus and prevent controllers from acting on that input

### PR DESCRIPTION
**Pull request by adam4813**
This will give the ability to click on a entity and have the entity send out a focus captured and blur (no longer has focus) event, and applies this event dispatch to the vcomputer.

## Description
There is now 2 new events FocusCaptured and FocusBlurred that indicate which entity has generated the event and which form of input that have captured or blurred.

The VComputer now listens for entity clicked events and if a click is on a computer with a keyboard it sends a FocusCaptured. Upon hitting the ESCAPE key an appropriate FocusBlurred event is sent out.

The simulation listens for these events and sets a flag on all controllers indicated if they have keyboard/mouse focus or not.

## Motivation and Context
Users need a way to interact with the computers, and there was no mechanism to stop the user from moving during input, nor was there a way to send input to a computer.

## How Has This Been Tested?
I connected to the server, moved to the computer with the black screen and AAAA on it in the world and clicked on it. After clicking on it I was able to see my characters appear on the computers screen. Clicking the other computer with the mostly red screen caused key to no longer appear on the first computer, but I was still unable to move around which proves that movement is lost when a computer has focus. Finally hitting the ESCAPE key caused movement to return and input to no longer be sent to the comptuer.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: common/vcomputer-system.cpp [hunk 1]
```diff
@@ -329,6 +360,11 @@ namespace tec {
 					Computer* computer = new Computer();
 					computer->In(comp);
 					ComputerComponentMap::Set(entity_id, computer);
+					for (auto device : computer->devices) {
+						if (device.second->device->DevType() == 0x03) { // 0x03 is keyboard DevType
```
Comment (Meisaka) [inline common/vcomputer-system.cpp:364]: `DevType()` is one of a group of methods designed for the emulation to identify devices.
Type 0x03 is "HID", which in theory can include things that aren't keyboards.
Using it this way will certainly work for now, but it isn't future proof.

**Review by Meisaka — APPROVED** [review]
I noticed this implementation uses booleans to keep track of focus, this will work, but it comes with drawbacks, just something to keep in mind.

File: common/vcomputer-system.cpp (hunk 1 above)
Comment (adam4813) [reply common/vcomputer-system.cpp:364]: I wasn't sure the best way to know if it should be able to receive focus. If there is a better way I will adjust accordingly.

**Comment by adam4813** [comment]
It is boolean for the time as it was the easiest way to get this out the door so I can focus on other aspects such as building. I agree it isn't a perfect solution and can get out of sync with several sources of truth.

Another factors is that we cannot explicitly stop the events at the "focused" entity because other elements might not care if it has focus and still need to process input such as the UI.

**Pull request merged**